helix-core/src/main/java/org/apache/helix/tools/ClusterSetup.java

```diff
@@ -189,6 +189,16 @@ public void addCluster(String clusterName, boolean overwritePrevious, CloudConfi
 
     if (cloudConfig != null) {
       _admin.addCloudConfig(clusterName, cloudConfig);
+      // If cloud is enabled and Cloud Provider is Azure, populated the Topology information in cluster config
+      if (cloudConfig.isCloudEnabled()
```

Comment:
1. Can we update the cloudconfig first then add it to the cluster? So that we have only one ZK access.
2. If the following config is required for AZURE, shall we add it to a more fundamental method instead of the tool? What if the user call the other API or admin method to create a cluster that is not compatible?

Reply:
1. Please note that we are updating two different paths. The new information will be added to cluster config. So eventually we need two accesses. one for the cloud config and one for the cluster config.
2. Our plan is to populate these information while creating the cluster and all of the paths (REST and java) to create cluster should pass through this part of the code.